src/read/page/stream.rs

```diff
@@ -64,7 +83,7 @@ fn _get_page_stream<R: AsyncRead + AsyncSeek + Unpin + Send>(
             if let Some(data_header) = data_header {
                 if !pages_filter(&descriptor, &data_header) {
                     // page to be skipped, we sill need to seek
-                    reader.seek(SeekFrom::Current(read_size)).await?;
+                    copy(reader.take(read_size as u64), &mut sink()).await?;
```

Comment:
won't this still perform a full read of the data?

Reply:
Yup, but I think `RangedAsyncReader` does this as well (copies the bytes into the internal buffer) or worse (needs to make a new HTTP request for the next read).

Reply:
It's worth noting our use-case doesn't filter ever. If you have a usecase where there is lots of filtering and additional HTTP requests are worthwhile because the amount of skipped data is large then would a `get_all_pages_stream` that just used a read be acceptable?

Reply:
If I understand correctly, your argument is that it is likely faster to pre-fetch 3 pages even if the middle one is skipped, than performing 2 requests (first and third page).

Under that argument, we should not even require `AsyncSeek`, right?

Reply:
> Under that argument, we should not even require AsyncSeek, right?

Yes *but* when I made that argument I was thinking of skipping one or two pages once in awhile. Also worth considering is say 1000 pages where all of the intermediate pages are filtered (leaving on the first and the last). I could see this benefiting from a second HTTP request rather than sinkholing the intermediate bytes.